src/infra/storage/storage.service.ts

```diff
@@ -1,32 +1,97 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, OnModuleInit } from '@nestjs/common';
+import { Client } from 'minio';
+import { randomUUID } from 'node:crypto';
+import { Stream } from 'stream';
+import * as Y from 'yjs';
 import { Logger } from '../logger/index.js';
+import { DocumentStorage } from '../y-redis/storage.js';
 import { StorageConfig } from './storage.config.js';
 
-@Injectable()
-export class StorageService {
-	private readonly bucketName: string;
+export const encodeS3ObjectName = (room: string, docid: string, r = ''): string =>
+	`${encodeURIComponent(room)}/${encodeURIComponent(docid)}/${r}`;
+
+const readStream = (stream: Stream): Promise<Buffer> =>
```

Comment:
Sollen die Methoden hier bleiben, oder private Methoden vom StorageService werden, oder in ein util File verschoben werden?